### DISCUSSION TITLE
Is getAuth callable multiple times?

### DISCUSSION THREAD

**Discussion by shivamCode0** [firebase/firebase-js-sdk#5254] [Q&A] [answered]
In *9.0.0-beta.8*, the new `getAuth` feature says that it "Initializes an Auth instance with platform specific default dependencies." Because it initializes a new instance, do I have to use a singleton pattern or can I call it multiple times? Even if I can, should I? Also, does it have the same effect as other `getX` methods?

**Comment by Feiyang1** [comment, marked as answer]
Yes, `getAuth` is idempotent and will return the existing instance. It only creates a new instance the first time it's called.

**Reply by shivamCode0** [reply]
Thanks! Can I have some info regarding this in the documentation or code to better understand how exactly it works? Also, would it be a good idea to add this info to the JSDoc? I think that it would make it clearer what `getAuth` does.

**Reply by Feiyang1** [reply]
Yup, this should fix it with the next doc update. https://github.com/firebase/firebase-js-sdk/pull/5255
